src/session.c

```diff
@@ -190,21 +222,23 @@ int save_session()
         int attempts = 0, ret_val = 0, i = 0;
 	struct wps_data *wps = NULL;
 	int pin_string;
+	int pin_empty;
```

Comment:
this stuff doesn't seem related to the functionality of this PR

please explain what it does in and open a separate PR for these changes

Reply:
I'm using globule->pin for store default pin to crack, if "empty" pin in the default pins list and it cracks the password, then pin_empty is for control to not save wpc file.
pin_string is for string mode with empty pin (-p ""), pin_empty is for any empty pin mode, either --genpin or --pin "".